lib/Screens/sign_up_screen.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:basic_authentication_flutter_challenge/Screens/home_screen.dart';
+import 'package:flutter/material.dart';
+import 'package:basic_authentication_flutter_challenge/Services/api.dart';
+
+class SignUpScreen extends StatefulWidget {
+  const SignUpScreen({Key? key}) : super(key: key);
+
+  @override
+  SignUpScreenState createState() => SignUpScreenState();
+}
+
+class SignUpScreenState extends State<SignUpScreen> {
+  TextEditingController userNameController = TextEditingController();
+  TextEditingController passwordController = TextEditingController();
+
+  @override
+  void dispose() {
+    userNameController.dispose();
+    passwordController.dispose();
+    super.dispose();
+  }
+
+  void showSnackBar(BuildContext context, String text) {
+    final snackBar = SnackBar(
+      content: Text(text),
+      duration: const Duration(seconds: 1),
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
+
+  void homePage() {
```

Comment:
It is good to move navigation code into its own function. 

You could also do this with `onPressed` function of Login button. Move all the code to a function named `login()`